src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -270,6 +270,36 @@ public new bool IsSectionDefined(string name)
             return zone != null && zone.Items.Count > 0;
         }
 
+        public new void DefineSection(string name, Microsoft.AspNetCore.Mvc.Razor.RenderAsyncDelegate section)
+        {
+            var zone = ThemeLayout.Zones[name];
+            dynamic sectionShape = New.AspSection(SectionContent: SectionDelegate(name,section) ).GetAwaiter().GetResult();
+
+            if (zone is Zones.ZoneOnDemand zoneOnDemand)
+            {
+                sectionShape = zoneOnDemand.AddAsync(sectionShape);
+            }
+            else
+            {
+                sectionShape = zone.Add(sectionShape);
+            }
+        }
+
+        private string SectionDelegate(string name, Microsoft.AspNetCore.Mvc.Razor.RenderAsyncDelegate oldDelegate)
+        {
+            string sectionContent = null;
+            Microsoft.AspNetCore.Mvc.Razor.RenderAsyncDelegate newDelegate = async () =>
+            {
+                var temporaryWriter = new System.IO.StringWriter();
```

Comment:
Maybe create a StringWriter() once and clear it before a new usage as mvc does

Reply:
Updated code to use private _writer member variable.